File: _includes/code/howto/manage-data.aliases.ts
import weaviate, { WeaviateClient } from 'weaviate-client'

const openaiKey = process.env.OPENAI_API_KEY as string

// START ConnectToWeaviate
// Connect to local Weaviate instance
const client: WeaviateClient = await weaviate.connectToLocal()
// END ConnectToWeaviate

// Cleanup
const aliases = await client.alias.listAll()
const collections = ["Articles", "ArticlesV2", "Products_v1", "Products_v2"]

if (aliases) {
    for (const item of aliases) {
        await client.alias.delete(item.alias)
    }
}

for (const collection of collections) {
    if (await client.collections.exists(collection)) {
        await client.collections.delete(collection)
    }
}


// START CreateAlias
// Create a collection first
await client.collections.create({
    name: "Articles",
    vectorizers: weaviate.configure.vectors.selfProvided(),
    properties: [
        { name: "title", dataType: weaviate.configure.dataType.TEXT },
        { name: "content", dataType: weaviate.configure.dataType.TEXT },
    ],
})

console.log('Created collection "Articles"')
// Create an alias pointing to the collection
await client.alias.create({
    alias: "ArticlesAlias",
    collection: "Articles"
})

console.log('Created alias "ArticlesAlias"')
// END CreateAlias

// START ListAllAliases
// Get all aliases in the instance
const allAliases = await client.alias.listAll()

if (allAliases) {
    for (const [_, aliasInfo] of Object.entries(allAliases)) {
        console.log(`Alias: ${aliasInfo.alias} -> Collection: ${aliasInfo.collection}`);
    }
}
// END ListAllAliases

// START ListCollectionAliases
// Get all aliases pointing to a specific collection
const collectionAliases = await client.alias.listAll({ collection: "Articles" })

if (collectionAliases) {
    for (const [_, aliasInfo] of Object.entries(collectionAliases)) {
        console.log(`Alias pointing to Articles: ${aliasInfo.alias}`);
    }
}

// END ListCollectionAliases

// START GetAlias
// Get information about a specific alias
const aliasInfo = await client.alias.get("ArticlesAlias")

if (aliasInfo) {
    console.log(`Alias: ${aliasInfo.alias}`);
    console.log(`Target collection: ${aliasInfo.collection}`);
}
// END GetAlias

// START UpdateAlias
// Create a new collection for migration
await client.collections.create({
    name: "ArticlesV2",
    vectorizers: weaviate.configure.vectors.selfProvided(),
    properties: [
        { name: "title", dataType: weaviate.configure.dataType.TEXT },
        { name: "content", dataType: weaviate.configure.dataType.TEXT },
        { name: "author", dataType: weaviate.configure.dataType.TEXT },  // New field
    ],
})

// Update the alias to point to the new collection
await client.alias.update({
    alias: "ArticlesAlias",
    newTargetCollection: "ArticlesV2"
})

console.log("Alias updated successfully")

// END UpdateAlias

await client.collections.delete("Articles")
// START UseAlias
// Ensure the Articles collection exists (it might have been deleted in previous examples)

await client.collections.create({
    name: "Articles",
    vectorizers: weaviate.configure.vectors.selfProvided(),
    properties: [
        { name: "title", dataType: weaviate.configure.dataType.TEXT },
        { name: "content", dataType: weaviate.configure.dataType.TEXT },
    ],
})
// END UseAlias

// START DeleteAlias
// Delete an alias (the underlying collection remains)
await client.alias.delete("ArticlesAlias")
// END DeleteAlias

// START UseAlias
// Create an alias for easier access
await client.alias.create({
    alias: "ArticlesAlias",
    collection: "Articles"
})

// Use the alias just like a collection name
const articles = client.collections.use("ArticlesAlias")

// Insert data using the alias
await articles.data.insert({
    "title": "Using Aliases in Weaviate",
    "content": "Aliases make collection management easier...",
})

// Query using the alias
const results = await articles.query.fetchObjects({ limit: 5 })

for (const obj of results.objects) {
    console.log(`Found: ${obj.properties['title']}`);
}
// END UseAlias

// START Step1CreateOriginal
// Create original collection with data
await client.collections.create({
    name: "Products_v1",
    vectorizers: weaviate.configure.vectors.selfProvided()
})

const products_v1 = client.collections.use("Products_v1")

await products_v1.data.insertMany([
    { "name": "Product A", "price": 100 },
    { "name": "Product B", "price": 200 }
])
// END Step1CreateOriginal

// START Step2CreateAlias
// Create alias pointing to current collection
await client.alias.create({
    alias: "ProductsAlias",
    collection: "Products_v1"
})

// END Step2CreateAlias

// START MigrationUseAlias
// Your application always uses the alias name "Products"
const prods = client.collections.use("ProductsAlias");

// Insert data through the alias
await prods.data.insert({ name: "Product C", price: 300 });

// Query through the alias
const res = await prods.query.fetchObjects({ limit: 5 });
for (const obj of res.objects) {
    console.log(`Product: ${obj.properties.name}, Price: $${obj.properties.price}`);
}
// END MigrationUseAlias

// START Step3NewCollection
// Create new collection with updated schema
await client.collections.create({
    name: "Products_v2",
    vectorizers: weaviate.configure.vectors.selfProvided(),
    properties: [
        { name: "name", dataType: weaviate.configure.dataType.TEXT },
        { name: "price", dataType: weaviate.configure.dataType.NUMBER },
        { name: "category", dataType: weaviate.configure.dataType.TEXT },  // New field
    ],
})
// END Step3NewCollection

// START Step4MigrateData
// Migrate data to new collection
const products_v2 = client.collections.use("Products_v2")
const oldData = (await products_v1.query.fetchObjects()).objects

for (const obj of oldData) {
    await products_v2.data.insert({
        "name": obj.properties["name"],
        "price": obj.properties["price"],
        "category": "General",  // Default value for new field
    })
}
// END Step4MigrateData

// START Step5UpdateAlias
// Switch alias to new collection (instant switch!)
await client.alias.update({
    alias: "ProductsAlias",
    newTargetCollection: "Products_v2"
})

// All queries using "ProductsAlias" alias now use the new collection
const products = client.collections.use("ProductsAlias")
const result = await products.query.fetchObjects({ limit: 1 })
console.log(result.objects[0].properties)  // Will include the new "category" field
// END Step5UpdateAlias

// START Step6Cleanup
// Clean up old collection after verification
await client.collections.delete("Products_v1")
// END Step6Cleanup

// Cleanup
const cleanUpAliases = await client.alias.listAll()

if (cleanUpAliases) {
    for (const item of cleanUpAliases) {
        await client.alias.delete(item.alias)
    }
}

for (const collection of collections) {
    if (await client.collections.exists(collection)) {
        await client.collections.delete(collection)
    }
}
await client.close()

File: _includes/code/howto/manage-data.aliases.py
import weaviate
import weaviate.classes as wvc

# Connect to Weaviate
client = weaviate.connect_to_local()

# Cleanup
client.alias.delete(alias_name="ArticlesProd")
client.alias.delete(alias_name="MyArticles")
client.alias.delete(alias_name="Products")
client.collections.delete("Articles")
client.collections.delete("ArticlesV2")
client.collections.delete("Products_v1")
client.collections.delete("Products_v2")

# START CreateAlias
# Create a collection first
client.collections.create(
    name="Articles",
    vector_config=wvc.config.Configure.Vectors.self_provided(),
    properties=[
        wvc.config.Property(name="title", data_type=wvc.config.DataType.TEXT),
        wvc.config.Property(name="content", data_type=wvc.config.DataType.TEXT),
    ],
)

# Create an alias pointing to the collection
client.alias.create(alias_name="ArticlesProd", target_collection="Articles")
# END CreateAlias

# START ListAllAliases
# Get all aliases in the instance
all_aliases = client.alias.list_all()

# Filter to show only aliases from this example
for alias_name, alias_info in all_aliases.items():
    if alias_info.collection in ["Articles", "ArticlesV2"]:
        print(f"Alias: {alias_info.alias} -> Collection: {alias_info.collection}")
# END ListAllAliases

# START ListCollectionAliases
# Get all aliases pointing to a specific collection
collection_aliases = client.alias.list_all(collection="Articles")

for alias_name, alias_info in collection_aliases.items():
    print(f"Alias pointing to Articles: {alias_info.alias}")
# END ListCollectionAliases

# START GetAlias
# Get information about a specific alias
alias_info = client.alias.get(alias_name="ArticlesProd")

if alias_info:
    print(f"Alias: {alias_info.alias}")
    print(f"Target collection: {alias_info.collection}")
# END GetAlias

# START UpdateAlias
# Create a new collection for migration
client.collections.create(
    name="ArticlesV2",
    vector_config=wvc.config.Configure.Vectors.self_provided(),
    properties=[
        wvc.config.Property(name="title", data_type=wvc.config.DataType.TEXT),
        wvc.config.Property(name="content", data_type=wvc.config.DataType.TEXT),
        wvc.config.Property(
            name="author", data_type=wvc.config.DataType.TEXT
        ),  # New field
    ],
)

# Update the alias to point to the new collection
success = client.alias.update(
    alias_name="ArticlesProd", new_target_collection="ArticlesV2"
)

if success:
    print("Alias updated successfully")
# END UpdateAlias

client.collections.delete("Articles")

# START UseAlias
# Ensure the Articles collection exists (it might have been deleted in previous examples)

client.collections.create(
    name="Articles",
    vector_config=wvc.config.Configure.Vectors.self_provided(),
    properties=[
        wvc.config.Property(name="title", data_type=wvc.config.DataType.TEXT),
        wvc.config.Property(name="content", data_type=wvc.config.DataType.TEXT),
    ],
)
# END UseAlias
# Delete alias if it exists from a previous run
client.alias.delete(alias_name="MyArticles")

# START DeleteAlias
# Delete an alias (the underlying collection remains)
client.alias.delete(alias_name="ArticlesProd")
# END DeleteAlias
# START UseAlias
# Create an alias for easier access
client.alias.create(alias_name="MyArticles", target_collection="Articles")

# Use the alias just like a collection name
articles = client.collections.use("MyArticles")

# Insert data using the alias
articles.data.insert(
    {
        "title": "Using Aliases in Weaviate",
        "content": "Aliases make collection management easier...",
    }
)

# Query using the alias
results = articles.query.fetch_objects(limit=5)

for obj in results.objects:
    print(f"Found: {obj.properties['title']}")

# Add a new property using the alias
articles.config.add_property(
    wvc.config.Property(name="author", data_type=wvc.config.DataType.TEXT)
)
# END UseAlias

# START MigrationExample
# Step 1: Create original collection with data
client.collections.create(
    name="Products_v1", vector_config=wvc.config.Configure.Vectors.self_provided()
)

products_v1 = client.collections.use("Products_v1")
products_v1.data.insert_many(
    [{"name": "Product A", "price": 100}, {"name": "Product B", "price": 200}]
)

# Step 2: Create alias pointing to current collection
client.alias.create(alias_name="Products", target_collection="Products_v1")

# Step 3: Create new collection with updated schema
client.collections.create(
    name="Products_v2",
    vector_config=wvc.config.Configure.Vectors.self_provided(),
    properties=[
        wvc.config.Property(name="name", data_type=wvc.config.DataType.TEXT),
        wvc.config.Property(name="price", data_type=wvc.config.DataType.NUMBER),
        wvc.config.Property(
            name="category", data_type=wvc.config.DataType.TEXT
        ),  # New field
    ],
)

# Step 4: Migrate data to new collection
products_v2 = client.collections.use("Products_v2")
old_data = products_v1.query.fetch_objects().objects

for obj in old_data:
    products_v2.data.insert(
        {
            "name": obj.properties["name"],
            "price": obj.properties["price"],
            "category": "General",  # Default value for new field
        }
    )

# Step 5: Switch alias to new collection (instant switch!)
client.alias.update(alias_name="Products", new_target_collection="Products_v2")

# All queries using "Products" alias now use the new collection
products = client.collections.use("Products")
result = products.query.fetch_objects(limit=1)
print(result.objects[0].properties)  # Will include the new "category" field

# Step 6: Clean up old collection after verification
client.collections.delete("Products_v1")
# END MigrationExample


# Cleanup
client.alias.delete(alias_name="MyArticles")
client.alias.delete(alias_name="Products")
client.alias.delete(alias_name="ArticlesProd")
client.collections.delete("Articles")
client.collections.delete("ArticlesV2")
client.collections.delete("Products_v1")
client.collections.delete("Products_v2")

client.close()
